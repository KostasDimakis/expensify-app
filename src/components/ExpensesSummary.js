import React from 'react';
import {connect} from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({total, count}) => (
    <p>
      Viewing {count} {count === 1 ? 'expense' : 'expenses'} totalling {numeral(
        total / 100).format('$0,0.00')}
    </p>
);

const mapStateToProps = (state) => ({
  count: state.expenses.length,
  total: selectExpensesTotal(state.expenses),
});

export default connect(mapStateToProps)(ExpensesSummary);

