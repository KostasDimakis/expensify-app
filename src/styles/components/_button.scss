.button {
  background: $blue;
  padding: $s-size;
  font-weight: 300;
  font-size: $font-size-large;
  color: white;
  border: none;
  display: inline-block;
  text-decoration: none;
  line-height: 1;

  &--link {
    background: none;
  }

  &--secondary {
    background: #888;
  }
}
