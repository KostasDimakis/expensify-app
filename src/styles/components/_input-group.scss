.input-group {
  margin-bottom: $m-size;
  display: flex;
  flex-direction: column;

  @media screen and (min-width: $desktop-breakpoint) {
    flex-direction: row;
    align-items: center;
    margin-bottom: $l-size;
  }

  &__item {
    margin-bottom: $s-size;

    @media screen and (min-width: $desktop-breakpoint) {
      margin-right: $m-size;
      margin-bottom: 0;
    }
  }
}
