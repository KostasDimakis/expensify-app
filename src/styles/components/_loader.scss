.loader {
  width: 100vw;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;

  &__image {
    height: 6rem;
    width: 6rem;
  }
}
