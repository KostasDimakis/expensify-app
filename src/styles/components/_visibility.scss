.show-for-mobile {
  @media screen and (min-width: $desktop-breakpoint) {
    display: none !important;
  }
}

.show-for-desktop {
  @media screen and (max-width: $desktop-breakpoint - .01rem) {
    display: none !important;
  }
}
