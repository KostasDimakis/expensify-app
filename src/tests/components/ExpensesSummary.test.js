import {ExpensesSummary} from '../../components/ExpensesSummary';
import React from 'react';
import {shallow} from 'enzyme';

it('should render correctly 1 expense', function() {
  const count = 1;
  const total = 86631;

  const wrapper = shallow(<ExpensesSummary count={count} total={total}/>);

  expect(wrapper).toMatchSnapshot();
});

it('should render correctly 2 expenses', function() {
  const count = 2;
  const total = 1000;

  const wrapper = shallow(<ExpensesSummary count={count} total={total}/>);

  expect(wrapper).toMatchSnapshot();
});
