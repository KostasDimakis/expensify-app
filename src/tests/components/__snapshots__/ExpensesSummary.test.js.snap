// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1 expense 1`] = `
<p>
  Viewing 
  1
   
  expense
   totalling 
  $866.31
</p>
`;

exports[`should render correctly 2 expenses 1`] = `
<p>
  Viewing 
  2
   
  expenses
   totalling 
  $10.00
</p>
`;
